python/grass/temporal/aggregation.py

```diff
@@ -360,11 +402,18 @@ def aggregate_by_topology(
                 # Create the r.series input file
                 filename = gscript.tempfile(True)
                 file = open(filename, "w")
-                for name in aggregation_list:
-                    string = "%s\n" % (name)
-                    file.write(string)
-                file.close()
-
+                if weighting:
+                    for i, name in enumerate(aggregation_list):
+                        string = name
+                        weight = aggregation_weights[i]
+                        file.write("%s|%f\n" % (string, weight))
+                    file.close()
```

Comment:
The file.close() call is duplicated in the branches and does not need to be. Since you are rewriting most of the file code here, do you want to just use with statement avoiding the need for explicit close() call?